### ISSUE TITLE
Изменение структуры базы данных

### ISSUE THREAD

**Issue by sborislav** [SB-MaterialAdmin/Web#122]
Изменение структуры базы данных:
1. Таблица _admins - уделание всех столбцов, которые связаны с сервером
2. Таблица _admins - остается персональныя информация и веб-группах
3. Изменение _admins_servers_groups - перенос столбцов из _admins для каждого сервера, добавление действия админки для каждого сервера
4. Веб-управление банами возможно, только есть соответствующий доступ к текущему серверу

**Comment by sborislav** [comment]
![sbbd](https://user-images.githubusercontent.com/22676634/35864793-d7c71e08-0b63-11e8-8f51-9a2395807c12.png)

[SB_mysql_create.txt](https://github.com/SB-MaterialAdmin/Web/files/1699423/SB_mysql_create.txt)
Не знаю только как улчше организовать работу админа CONSOLE

**Comment by tallanvor** [comment]
Меня одного такая тавтология в названиях таблиц вгоняет в ступор?

2018-02-06 16:34 GMT+02:00 sborislav <notifications@github.com>:

**Comment by sborislav** [comment]
Согласен, понять что куда следует, если ты это не проектировал это, очень сложно. Поэтому и предлагаю изменить структуру

**Comment by tallanvor** [comment]
так я думал, это и есть предложенная схема
и как раз в ней - жесткая тавтология

6 февр. 2018 г. 5:49 ПП пользователь "sborislav" <notifications@github.com>
написал:

Согласен, понять что куда следует, если ты это не проектировал это, очень
сложно. Поэтому и предлагаю изменить структуру

—
You are receiving this because you commented.

Reply to this email directly, view it on GitHub
<https://github.com/SB-MaterialAdmin/Web/issues/122#issuecomment-363464938>,
or mute the thread

**Comment by sborislav** [comment]
Названия таблиц и колонки взяты из текущей структуры, чтобы разработчикам было проще понять что предлагается изменить.

**Comment by CrazyHackGUT** [comment]
Наконец-то, дошли руки посмотреть с нормального домашнего дисплея эту структуру.
И пока только один вопрос.
Что за `sgid`, и почему на него ссылаются баны и муты? Уникальный идентификатор "разрешения"?
Так же не ясно, зачем `maxbantime` и `maxmutetime` предлагается перенести в `admins_servers_groups`. Это фича чисто для групп. Если выставлять её каждому юзеру отдельно - можно задолбаться.
Так же имеет смысл снести `expired` из `admins`.

**Comment by sborislav** [comment]
1. _**sgid**_ условно говоря да, уникальный идеинтификатор прав на конкретном сервере. До этого баны и муты определялись по **_id серверу_** и **_id админа_**. Кстате, в это структуре есть один, но большой **НО**, если у пользователя права на группу серверов, значит мы не сможем узнать где его забанили( придется возвращать столбец с **_id сервера_**.

2. Что касается максимального времени мута и бана, то эти параметры так же как и флаги должны иметь более высокий приоретет и дополнять пользователя. В таком случае, чтобы не захламлять основную таблицу можно создать дополнительную и перенести все туда и заполнять её когда у пользлвателя есть дополнительные права.
3. Истечение веб прав предполагается, чтобы админка тоже имела время доступа, чтобы можно быстро забанить админа "-1" и он не смог зайти на сайт. До этого, вообще не было возможности отключения админок, только если снять права на сервер или группу.
4. Кстати, когда я держал сервер меня многие просили замораживать их админки, если эта фича актуальна, то можно её как-то продумать.
